bach/bach/dataframe.py

```diff
@@ -1994,18 +2023,24 @@ def to_pandas(self, limit: Union[int, slice] = None) -> pandas.DataFrame:
             This function queries the database.
         """
         sql = self.view_sql(limit=limit)
+        dialect = self.engine.dialect
 
         series_name_to_dtype = {}
-        for series in self.all_series.values():
+        for name, series in self.all_series.items():
             pandas_info = series.to_pandas_info()
             if pandas_info is not None:
-                series_name_to_dtype[series.name] = pandas_info.dtype
+                sql_column_name = get_sql_column_name(dialect=dialect, name=name)
+                series_name_to_dtype[sql_column_name] = pandas_info.dtype
 
         with self.engine.connect() as conn:
             # read_sql_query expects a parameterized query, so we need to escape the parameter characters
             sql = escape_parameter_characters(conn, sql)
             pandas_df = pandas.read_sql_query(sql, conn, dtype=series_name_to_dtype)
 
+        # Rename the columns that the query gives, to the actual column names of the DataFrame
```

Comment:
IMHO we should avoid mixing `columns` and `series` words (even for Pandas)
```suggestion
        # Rename the columns that the query gives, to the actual series names of the DataFrame
```

Reply:
Fixed in 51398ace5bada07f9cd84b7a2546e65afb2f7327